### ISSUE TITLE
ErrByzantineData includes bad shares

### ISSUE THREAD

**Issue by musalbas** [celestiaorg/rsmt2d#112]
It seems that ErrByzantineData also includes bad shares _after_ reconstruction, when I believe the expected behavior (as expected by celestia-node) is that it should only return shares that are known to be good (eg Merkle proofs for them are known).

**Comment by musalbas** [comment]
No. Let's assume you have proven shares for 5 cells in a row consisting of 10 cells. Then you try to extend it, and then you find out that once you compute the Merkle root, it does not match the committed row root, because those newly repaired cells are _not_ committed to in the root.

**Comment by Wondertan** [comment]
Ah, so "bad shares" are those newly repaired ones. I assumed these are good shares because these are shares that should've been committed while they aren't.

**Comment by Wondertan** [comment]
After a deep discussion with @musalbas in sync, we concluded that this issue could cause celestia-node to stall when incorrect committed data is withheld. 

In [the code](https://github.com/celestiaorg/celestia-node/blob/main/ipld/get.go#L67-L78), the rebuilt shares are passed as non-nil [causing a network request](https://github.com/celestiaorg/celestia-node/blob/main/ipld/get.go#L71) for respective(same index) incorrect shares to be retrieved, which might not be available, causing the node to wait forever or stall. This might be fixed on the node side by setting the timeout or on the rsmt2d side(this issue), which is preferred.
